core/base/src/TUrl.cxx

```diff
@@ -584,14 +586,15 @@ TObjArray *TUrl::GetSpecialProtocols()
    if (usedEnv)
       return fgSpecialProtocols;
 
+   R__LOCKGUARD(gROOTMutex);
    if (fgSpecialProtocols)
       fgSpecialProtocols->Delete();
 
    if (!fgSpecialProtocols)
       fgSpecialProtocols = new TObjArray;
 
    const char *protos = gEnv->GetValue("Url.Special", "file: hpss: dcache: dcap:");
-   usedEnv = kTRUE;
+   usedEnv = true;
```

Comment:
Need to be move to the very end to avoid a 2nd thread used partially initialized state.

Reply:
Doesn't the `R__LOCKGUARD(gROOTMutex);` a few lines above (line 589 in the new file) prevent that?